x-pack/plugins/observability_solution/observability_ai_assistant/server/service/kb_component_template.ts

```diff
@@ -32,6 +32,15 @@ export const kbComponentTemplate: ClusterComponentTemplate['component_template']
       '@timestamp': date,
       id: keyword,
       doc_id: { type: 'text', fielddata: true },
+      title: {
+        type: 'text',
+        fields: {
+          keyword: {
+            type: 'keyword',
+            ignore_above: 256,
+          },
```

Comment:
Adding nested keyword in order to be able to sort on it. Using nested keyword is recommended over `fielddata` as it is more performant.